src/main/java/com/google/devtools/build/lib/remote/TreeNodeRepository.java

```diff
@@ -377,6 +432,13 @@ private Digest actionInputToDigest(ActionInput input) throws IOException {
     if (input instanceof VirtualActionInput) {
       return Preconditions.checkNotNull(virtualInputDigestCache.get(input));
     }
+    Metadata metadata = Preconditions.checkNotNull(inputFileCache.getMetadata(input));
+    byte[] digest = metadata.getDigest();
+    if (digest == null) {
+      // If the artifact does not have a digest, it is because it is a directory.
+      // We get the digest from the set of Merkle hashes computed in this TreeNodeRepository.
+      return Preconditions.checkNotNull(inputDirectoryDigestCache.get(input));
```

Comment:
can you please add a message stating what went wrong. makes errors easier to debug.